record-and-playback/core/lib/recordandplayback/generators/events.rb

```diff
@@ -125,57 +177,203 @@ def self.create_webcam_edl(events, archive_dir)
+            stream = event.at_xpath('stream').text
+            filename = "#{video_dir}/#{stream}.flv"
+          when 'StartWebRTCShareEvent', 'StopWebRTCShareEvent'
+            uri = event.at_xpath('filename').text
+            filename = "#{video_dir}/#{File.basename(uri)}"
+          end
+          raise "Couldn't determine webcam filename" if filename.nil?
+  
+          # Add the video to the EDL
+          case event['eventname']
+          when 'StartWebcamShareEvent', 'StartWebRTCShareEvent'
+            videos[filename] = { :timestamp => timestamp }
+            active_videos << filename
+  
+            edl_entry = {
+              :timestamp => timestamp,
+              :areas => { :webcam => [] }
+            }
+            active_videos.each do |filename|
+              edl_entry[:areas][:webcam] << {
+                :filename => filename,
+                :timestamp => timestamp - videos[filename][:timestamp]
+              }
+            end
+            video_edl << edl_entry
+          when 'StopWebcamShareEvent', 'StopWebRTCShareEvent'
+            active_videos.delete(filename)
+  
+            edl_entry = {
+              :timestamp => timestamp,
+              :areas => { :webcam => [] }
+            }
+            active_videos.each do |filename|
+              edl_entry[:areas][:webcam] << {
+                :filename => filename,
+                :timestamp => timestamp - videos[filename][:timestamp]
+              }
+            end
+            video_edl << edl_entry
+          end
         end
-        raise "Couldn't determine webcam filename" if filename.nil?
+      else
+        events.xpath('/recording/event[@module="WEBCAM" or (@module="bbb-webrtc-sfu" and (@eventname="StartWebRTCShareEvent" or @eventname="StopWebRTCShareEvent")) or (@module="PARTICIPANT" and (@eventname="ParticipantStatusChangeEvent" or @eventname="ParticipantJoinEvent")) or @eventname="WebcamsOnlyForModeratorEvent" or @eventname="MeetingConfigurationEvent"]').each do |event|
+          timestamp = event['timestamp'].to_i - initial_timestamp
+
+          # Determine the video filename if event is as the following
+          case event['eventname']
+          when 'StartWebcamShareEvent', 'StopWebcamShareEvent'
+            stream = event.at_xpath('stream').text
+            filename = "#{video_dir}/#{stream}.flv"
+          when 'StartWebRTCShareEvent', 'StopWebRTCShareEvent'
+            uri = event.at_xpath('filename').text
+            filename = "#{video_dir}/#{File.basename(uri)}"
+          end
 
-        # Add the video to the EDL
-        case event['eventname']
-        when 'StartWebcamShareEvent', 'StartWebRTCShareEvent'
-          videos[filename] = { :timestamp => timestamp }
-          active_videos << filename
+          # Add the video to the EDL
+          case event['eventname']
+          when 'StartWebcamShareEvent', 'StartWebRTCShareEvent'
+            userId = BigBlueButton::Events.get_id_from_filename(filename)
```

Comment:
I thought it was requested that the userId event should be added to the share/stop events so that we would not need to use the filename as the main method of associating a camera stream to an userId.

If that was the case, this should be the fallback (in case the scripts are run over older recordings), not the default method. What's the motivation here?

Reply:
Yeah, so during implementation, I realized that even if I had the `userId` at the start/stop events, that wouldn't really mean that the EDL entry item has that `userId`, for example: There might be cameras already open when the start/stop events are interpreted, and they, surely, don't have the same `userId` as the newly added.

And about the motivation, I thought that by putting the userId in the EDL, we could use it in the future with new features, but I can take it out.